src/index.js

```diff
@@ -55,6 +55,19 @@ const expandShorthands = (tr) => {
 }
 
 /**
+ * Transforms __ref to ref
+ * @param {object} props
+ * @returns {object}
+ * @private
+ */
+const transformRef = ({ __ref, ...props }) => ({ ref: __ref, ...props })
```

Comment:
A) It should only do this if `__ref` was present.
B) I would move it to a separate file.